### PR TITLE
[BAEL-5211] - How to test a Java Optional

### PR DESCRIPTION
**Pull request by lucaCambi77**
A few tests on how to test a Java Optional

### CODE REVIEW

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OptionalUnitTest {
+
+    private final String stringToTest = "testString";
```
Comment (ashleyfrieze) [inline testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:14]: This should probably be `private static final String EXPECTED_STRING` - in that there's no instance-level aspect to it.

Similarly, this isn't the string we're testing - it's the result we're expecting in our tests.

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OptionalUnitTest {
```
Comment (ashleyfrieze) [inline testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:12]: By convention JUnit 5 tests don't have `public` modifiers on methods or the class

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Test
+    public void givenEmptyOptional_whenAssertIsNotPresent_thenOk() {
+        Optional<String> optional = optionalIsEmptyMethodToTest();
+        assertFalse(optional.isPresent());
+    }
+
+    private Optional<String> optionalWithValueMethodToTest() {
+        return Optional.of(stringToTest);
+    }
```
Comment (ashleyfrieze) [inline testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:64]: You've `DRY`ed out the tests by doing extract method here of a common bit of data, but it makes the test harder to follow. Inline these. In general, a trivial bit of code like this is easier re-written every time in a test, and that's especially true for example snippets in articles.

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @Test
+    public void givenOptionalWithValue_whenAssertEqualsGet_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+        assertEquals(stringToTest, optional.get());
+    }
+
+    @Test
+    public void givenOptionalWithValue_whenAssertTrueIsPresentAndGet_thenOk() {
+        Optional<String> optional = optionalWithValueMethodToTest();
+        assertTrue(optional.isPresent() && stringToTest.equals(optional.get()));
```
Comment (ashleyfrieze) [inline testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:40]: This is a bad test - it's a bad bit of advice. If this fails, we don't know why.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java (hunk 4 above)
Comment (lucaCambi77) [reply testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:40]: yes now that i see it i agree

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java (hunk 3 above)
Comment (lucaCambi77) [reply testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:64]: I agree but I was on to follow the StackOverflow example

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java (hunk 2 above)
Comment (lucaCambi77) [reply testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:12]: yes that is true

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java (hunk 1 above)
Comment (lucaCambi77) [reply testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:14]: ok i'll change it accordingly

File: testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java (hunk 3 above)
Comment (ashleyfrieze) [reply testing-modules/testing-assertions/src/test/java/com/baeldung/optional/OptionalUnitTest.java:64]: Let's make our own version of the example that suits our needs.

SO examples aren't always the best ideas ;)

**Pull request merged**